src/environment/environment.py

```diff
@@ -0,0 +1,165 @@
+import pygame
+import numpy as np
+from pygame.surface import Surface
+from typing import Tuple
+
+
+class Environment:
+    def __init__(
+        self, screen: Surface, agent_size: int = 10, render_on: bool = False
+    ) -> None:
+        """
+        Initializes the environment.
+
+        Args:
+            screen (Surface): Pygame screen surface.
+            agent_size (int, optional): Size of the agent. Defaults to 10.
+            render_on (bool, optional): Whether to render the environment. Defaults to False.
+        """
+        self.screen = screen
+        self.render_on = render_on
+        self.agent_size = agent_size
+
+        self.agent_location = (
+            (self.screen.get_size()[0] / 2),
+            (self.screen.get_size()[1] / 2),
+        )
+
+        self.rewards = {
+            "alive": 1,
+            "death": -100,
+            "not possible": -5,
+        }
+
+        self.actions = {
+            "up": 0,
+            "down": 1,
+            "left": 2,
+            "right": 3,
+        }
+
+    def reset(self) -> np.ndarray:
+        """
+        Resets the environment.
+
+        Returns:
+            np.ndarray: Initial state of the environment.
+        """
+        self.agent_location = (
+            (self.screen.get_size()[0] / 2),
+            (self.screen.get_size()[1] / 2),
+        )
+
+        # TODO: reset enemies; self.enemy_generator.reset()? Dunno if needed
+
+        if self.render_on:
+            self.render()
+
+        return self.get_state()
+
+    def render(self) -> None:
+        """
+        Renders the environment.
+        """
+        self.screen.fill((0, 0, 0))
+        pygame.draw.circle(
+            self.screen, (255, 255, 255), self.agent_location, self.agent_size
+        )
+        # TODO: draw enemeis; self.enemies.render()?
+        pygame.display.flip()
+
+    def get_state(self) -> np.ndarray:
+        """
+        Gets the current state of the environment.
```

Comment:
Maybe add some explanation on how the state looks like, e.g.:

```suggestion
        Gets the current state of the environment. This state is a list of coordinates, first the agents coordinates followed by the enemy coordinates, e.g. [400, 300, 10, 10].
```